starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +20,43 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord (array) {
+  var longest = 0; 
+  for (var i = 0; i < array.length; i++) {
+    if (array[i].length > longest) {
+      longest = array[i].length;
+    } 
+      return longest;
+    }
+  } 
+  
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array) {
+  if (array !== "") {
+  var result=array.reduce(function(a,b){
+  return a + b;  
+  })
+} else {
+  return 0;
+}
+};
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array){
```

Comment:
Os pasa lo mismo que en de arriba, también tenéis que controlar si el la variable result tiene valor para poder dividirla, porque sino os darían error.